### PR TITLE
Rename mimirpb.INVALID error cause into mimirpb.UNKNOWN_CAUSE

### PR DESCRIPTION
**Pull request by duricanikolic**
#### What this PR does
This PR renames the `mimirpb.INVALID` symbolic value of type `mimirpb.ErrorCause` introduced in https://github.com/grafana/mimir/pull/6377 into `mimirpb.UNKNOWN_CAUSE`. The reason for this is that `mimirpb.INVALID` could be misinterpreted as something that is not valid, but its actual purpose is to represent an unknown error cause.

#### Which issue(s) this PR fixes or relates to

Relates to https://github.com/grafana/mimir/issues/6008

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/errors_test.go [hunk 1]
```diff
@@ -709,3 +709,43 @@ func checkErrorWithStatusDetails(t *testing.T, details []any, expectedDetails *m
 		require.Equal(t, expectedDetails, errDetails)
 	}
 }
+
```
Comment (56quarters) [inline pkg/ingester/errors_test.go:712]: nit: Feels a bit odd to put methods from a non-test type into the `_test.go` file. Can you add a comment explaining that this is only needed for tests?

**Review by 56quarters — APPROVED** [review]
LGTM, thanks!

File: pkg/ingester/errors_test.go (hunk 1 above)
Comment (duricanikolic) [reply pkg/ingester/errors_test.go]: I have reverted this change, and put the methods back to `errors.go` with a comment stating that these methods are needed for tests only.

**Comment by colega** [comment]
I can't agree with this change.

> The reason for this is that mimirpb.INVALID could be misinterpreted as something that is not valid, but its actual purpose is to represent an unknown error cause.

The reason why I suggested to have an `mimirpb.INVALID` is to avoid having a valid zero value, that could easily lead to bugs. 

With this new version of the code it is unclear when the code is buggy and it didn't set the reason, and when the error was processed and it falls into the "unknown" case. 

I'm also not sure when the cause is "Unknown". It seems that we set this into the details of "Unavailable" errors, but maybe in those cases we just shouldn't set any details at all?

**Pull request merged**